File: apps/php-sdk/src/Models/AgentTraceError.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentTraceError
{
    public function __construct(
        private readonly ?string $code = null,
        private readonly ?string $source = null,
        private readonly bool $retryable = false,
        private readonly ?string $message = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            code: $data['code'] ?? null,
            source: $data['source'] ?? null,
            retryable: (bool) ($data['retryable'] ?? false),
            message: $data['message'] ?? null,
        );
    }

    public function getCode(): ?string
    {
        return $this->code;
    }

    public function getSource(): ?string
    {
        return $this->source;
    }

    public function isRetryable(): bool
    {
        return $this->retryable;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }
}

File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentTraceError.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Structured error carried by agent trace events.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentTraceError {

    private String code;
    private String source;
    private boolean retryable;
    private String message;

    public String getCode() { return code; }
    public String getSource() { return source; }
    public boolean isRetryable() { return retryable; }
    public String getMessage() { return message; }

    @Override
    public String toString() {
        return "AgentTraceError{code=" + code + ", source=" + source + ", retryable=" + retryable + "}";
    }
}
